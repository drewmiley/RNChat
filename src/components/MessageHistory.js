import React, { Component } from 'react';
import { View } from 'react-native';

import MessageView from './MessageView';

export default class MessageHistory extends Component {
    constructor(props) {
        super(props);
        this.shouldComponentUpdate = function(nextProps, nextState) {
            return this.props.messageHistory !== nextProps.messageHistory;
        };
    };
    render() {
        return <View>
            {this.props.messageHistory.map(message =>
                <MessageView key={message.get('id')}
                    drewSaid={message.get('drewSaid')}
                    text={message.get('text')} />
            )}
        </View>
    };
};