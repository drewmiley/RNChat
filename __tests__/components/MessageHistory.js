import 'react-native';
import { List, Map } from 'immutable';
import React from 'react';
import renderer from 'react-test-renderer';

import MessageHistory from '../..//src/components/MessageHistory';

describe('MessageHistory', () => {

    describe('Rendering', () => {

        it('should match the snapshot', () => {
            const messageHistory = List.of(
            	Map({
	            	id: 1,
	            	drewSaid: true,
	            	text: 'Hello'
            	}),
            	Map({
	            	id: 2,
	            	drewSaid: false,
	            	text: 'Hello'
            	})
            );
            const component = renderer.create(
                <MessageHistory  messageHistory={messageHistory} />
            );            
            expect(component).toMatchSnapshot();
        });

    });

});