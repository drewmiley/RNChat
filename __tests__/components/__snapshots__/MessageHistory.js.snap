exports[`MessageHistory Rendering should match the snapshot 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    DREW : 
    Hello
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    YOU : 
    Hello
  </Text>
</View>
`;
